takari-lifecycle-plugin/src/main/java/io/takari/maven/plugins/compile/jdt/ProcessingEnvImpl.java

```diff
@@ -112,11 +112,11 @@ public Factory getFactory() {
     return _factory;
   }
 
-  public void hardReset() {
+  public void incrementalIterationReset() {
     reset();
     setErrorRaised(false);
 
-    ((FilerImpl) _filer).hardReset();
+    ((FilerImpl) _filer).incrementalIterationReset();
```

Comment:
Yeah this should be documented on the Filer.   seems to be reset between incremental compiles, but not between rounds.